src/server_manager/infrastructure/repository.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export type KeyExtractor<Record, Key> = (r: Record) => Key;
+export type KeyComparator<Key> = (k1: Key, k2: Key) => boolean;
+
+export class LocalStorageRepository<Record, Key> {
```

Comment:
"Repository" is a meaningful word in DDD, and using it here may sound like a class Repositories should extend. Perhaps just call it Storage or LocalStorage? This is really about persistence.

Reply:
Yeah, this is purely a persistence class. It's part of our `infrastructure` classes and I agree, calling it a repository makes it confusing. I've changed it to KeyValueStorage.